cmd/query/app/http_handler_test.go

```diff
@@ -623,6 +623,38 @@ func TestGetOperationsLegacyStorageFailure(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestTransformOTLPSuccess(t *testing.T) {
+	withTestServer(func(ts *testServer) {
+		response := new(interface{})
+		request := new(interface{})
+
+		requestBytes := readOTLPTraces(t)
+
+		err := json.Unmarshal(requestBytes, request)
+		require.NoError(t, err)
+
+		err = postJSON(ts.server.URL+"/api/transform", request, response)
+		require.NoError(t, err)
+
+		corectResponseBytes := readJaegerTraces(t)
+		corectResponse := new(interface{})
+
+		err = json.Unmarshal(corectResponseBytes, corectResponse)
```

Comment:
none of this marshaling is required for the test. You need to read a fixture as bytes, submit them to the server, receive back bytes, and compare them with the expected fixture.